test/unit/src/test/java/javax/faces/component/SearchExpressionHandlerTest.java

```diff
@@ -0,0 +1,997 @@
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:3:nested:1:nestedText"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:3:baseText"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:0:baseText"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:3:nested:0:nestedText"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:3:nested"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "mainForm:table:1:nested:0:nestedText"));
+        
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@this"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@this:@parent:showName"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@parent:showName:@parent:showName"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@form:showName"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@namingcontainer:showName"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@previous"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@next"));
+        Assert.assertTrue(handler.isValidExpression(searchExpressionContext, "@parent:@id(msgName)"));
+        
+        Assert.assertFalse(handler.isValidExpression(searchExpressionContext, "@whoNows"));
+        Assert.assertFalse(handler.isValidExpression(searchExpressionContext, "@parent:@whoNows"));
+        Assert.assertFalse(handler.isValidExpression(searchExpressionContext, "mainForm:@whoNows"));
+        
+        Assert.assertFalse(handler.isValidExpression(searchExpressionContext, "@none:@parent"));
+        Assert.assertFalse(handler.isValidExpression(searchExpressionContext, "@all:@parent"));
+    }
+    
+    public void test_ResolveComponents_Id() {
+		UIComponent root = new UIPanel();
+        FacesContext.getCurrentInstance().getViewRoot().getChildren().add(root);
+
+		UINamingContainer outerContainer = new UINamingContainer();
+		outerContainer.setId("myContainer");
+		root.getChildren().add(outerContainer);
+
+		UIForm form = new UIForm();
+		form.setId("form");
+		root.getChildren().add(form);
+
+		UINamingContainer innerContainer = new UINamingContainer();
+		innerContainer.setId("myContainer");
+		form.getChildren().add(innerContainer);
+
+		UINamingContainer innerContainer2 = new UINamingContainer();
+		innerContainer2.setId("myContainer2");
+		form.getChildren().add(innerContainer2);
+
+        List<UIComponent> result = resolveComponents(form, " @id(myContainer) ");
+        assertTrue(result.size() == 1);
+        assertTrue(result.contains(innerContainer));
+    }
+    
+    public void test_PFS() {
```

Comment:
That it's referring to "PrimeFaces Selectors".

Not a big problem for now as this is "just" an IT. But perhaps a future maintainer might wonder what exactly you wanted to test here.